webapp/components/sidebar.jsx

```diff
@@ -382,11 +383,13 @@ export default class Sidebar extends React.Component {
     }
 
     sortChannelsByDisplayName(a, b) {
+        const locale = LocalizationStore.getLocale();
+
         if (a.display_name === b.display_name) {
-            return a.name.localeCompare(b.name);
+            return a.name.localeCompare(b.name, locale, {numeric: true, caseFirst: 'upper'});
```

Comment:
Oh that's interesting. I didn't know those options existed. I'll have to keep those in mind when using `localeCompare` in other places.

Maybe just take out the `caseFirst` option though? I don't know what the standard for sorting upper vs lower case is, but the default appears to just figure it out from the locale.